src/main/java/org/folio/consortia/service/impl/UserAffiliationServiceImpl.java

```diff
@@ -78,6 +79,44 @@ public void createPrimaryUserAffiliation(String eventPayload) {
     }
   }
 
+  @Override
+  @SneakyThrows
+  @Transactional
+  public void updatePrimaryUserAffiliation(String eventPayload) {
+    FolioExecutionContext currentContext = (FolioExecutionContext) folioExecutionContext.getInstance();
+    String centralTenantId = folioExecutionContext.getTenantId();
+    try {
+      var userEvent = objectMapper.readValue(eventPayload, UserEvent.class);
+      log.info("Received event for update primary affiliation for user: {} and tenant: {}", userEvent.getUserDto().getId(), userEvent.getTenantId());
+
+      var consortiaTenant = tenantService.getByTenantId(userEvent.getTenantId());
+      if (consortiaTenant == null) {
+        log.warn("Tenant {} not exists in consortia", userEvent.getTenantId());
+        return;
+      }
+      UUID userId = getUserId(userEvent);
+      String newUsername = userEvent.getUserDto().getUsername();
+
+      UserTenantCollection userTenantCollection = userTenantService.getByUserId(consortiaTenant.getConsortiumId(), userId, 0, Integer.MAX_VALUE);
+      boolean isUsernameChanged = userTenantCollection.getUserTenants().stream()
+        .anyMatch(userTenant -> ObjectUtils.notEqual(userTenant.getUsername(), newUsername));
+
+      if (isUsernameChanged) {
```

Comment:
Just help for reviewers, we are not storing email, phoneNumber, mobilePhoneNumber in mod-consortia, so username is only field that can be updated.
But we should send all fields for CONSORTIUM_PRIMARY_AFFILIATION events to populate in mod-users central tenant